File: src/main/java/block/BlockLog.java
package block;

import gl.UVPair;
import registry.RegistryName;

public class BlockLog extends Block {
    private LogType _type;

    public BlockLog(LogType type) {
        super(new RegistryName("blockgame", type.getTypeName()+"_log"));
        _type = type;
    }

    @Override
    public UVPair getUVForFace(EnumDirection face) {
        switch(face) {
            case UP:
            case DOWN:
                return _type.getTop();
            default:
                return _type.getSide();
        }
    }
}

File: src/main/kotlin/block/BlockLog.kt
package block

import gl.UVPair
import registry.RegistryName

class BlockLog(val type: LogType) : Block(RegistryName("blockgame", "${type.type_name}_log")) {

    override fun getUVForFace(face: EnumDirection): UVPair {
        return when (face) {
            EnumDirection.UP, EnumDirection.DOWN -> type.top
            else -> type.side
        }
    }
}